### PR TITLE
[improvement] Migrate all UT cases from jUnit4 to jUnit5 in dao module

### LINKED ISSUE #12319: Migrate all UT cases from jUnit 4 to jUnit 5 in dao module
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

- Migrate all UT cases from jUnit 4 to jUnit 5 in ```dao ```module.
- This is a part of #12301

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuchanghai**
## Brief change log

- Migrate all UT cases from jUnit4 to jUnit5 in dao module
- Related: #12301 
- Close Issue: #12319 
 
this close #12319

### CODE REVIEW

**Comment by fuchanghai** [comment]
@EricGao888

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/AlertDaoTest.java [hunk 1]
```diff
@@ -23,8 +23,8 @@
 
 import java.util.List;
 
-import org.junit.Assert;
-import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
 import org.junit.runner.RunWith;
```
Comment (EricGao888) [inline dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/AlertDaoTest.java]: We need to remove this, use `ExtendWith` instead as stated in #12301 SOP.

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/upgrade/WorkerGroupDaoTest.java [hunk 2]
```diff
@@ -25,7 +25,8 @@
 
 import javax.sql.DataSource;
 
-import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
 import org.junit.runner.RunWith;
```
Comment (EricGao888) [inline dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/upgrade/WorkerGroupDaoTest.java]: Same as above

**Review by EricGao888 — CHANGES_REQUESTED** [review]

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/upgrade/WorkerGroupDaoTest.java (hunk 2 above)
Comment (fuchanghai) [reply dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/upgrade/WorkerGroupDaoTest.java]: done

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/AlertDaoTest.java (hunk 1 above)
Comment (fuchanghai) [reply dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/AlertDaoTest.java]: done

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/upgrade/WorkerGroupDaoTest.java [hunk 3]
```diff
@@ -47,14 +45,15 @@
 
         Map<Integer, String> workerGroupMap = workerGroupDao.queryAllOldWorkerGroup(dataSource.getConnection());
 
-        assertThat(workerGroupMap.size(), greaterThanOrEqualTo(0));
+        Assertions.assertTrue(workerGroupMap.size() >= 0);
```
[the remark this answers is not in the document]
Comment (fuchanghai) [reply dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/upgrade/WorkerGroupDaoTest.java]: done

[the remark this answers is not in the document]
Comment (fuchanghai) [reply dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/upgrade/WorkerGroupDaoTest.java]: done

**Comment by EricGao888** [comment]
Seems network problems, have rerun the failed CI.

**Review by EricGao888 — APPROVED** [review]
LGTM, thanks

**Pull request merged**